articles/network-routing/index.md

```diff
@@ -0,0 +1,145 @@
+### Understanding Network Routing
+
+![Routing](/engineering-education/network-routing/routing.jpeg)
+
+When using the internet, our devices send requests to servers stored in various data centers in packets. Likewise, the servers return responses to the requests using data packets.</br>
+These packets' journey to the data center from our devices and vice versa form the internet's backbone.</br>
+Nevertheless, controlling these packets from the data source to the destination through the wide complicated global network is not a walk in the park.</br>
+This is where routing comes in. It is done by specialized networking hardware called **routers**.</br>
+Routers select an appropriate path that will ensure the packets arrive faster and safely.</br>
+
+Take an example of when one wants to get home situated in an estate from his/her workplace located in a city center. During rush hours, cities are usually characterized by traffic snarl-ups on their roads. The person will then use Google maps to check the route with less traffic and maneuver all the way home.</br>
+Similarly, routers use algorithms to make logical data decisions for selecting appropriate paths for forwarding the packets. The decisions are made using the current network states of where the packets would pass through.
+
+There are two types of routing:</br>
+- **Static**</br>
+- **Dynamic**</br>
+
+In **static routing**, *the routing tables information don't change after being manually set up* by a network administrator. If a network breaks in the link, this information doesn't change unless the change is done manually.</br>
+This is the exact opposite of **dynamic routing** where *the tables update themselves dynamically in discrete time steps according to the current network states* such as link failures, traffic changes, etc.</br>
+A commonly used algorithm in dynamic routing is the **Link State algorithm**. It checks the states of the links between nodes and makes forwarding decisions.</br>
+Let's discuss it.</br>
+**Link State algorithm**</br>
+Link state consists of two algorithms:</br>
```

Comment:
Arent these two types of the link-state algorithm?

Reply:
Yeah, they are under the Link state algorithm. I will rectify the readability

Reply:
Rectified